### PR TITLE
Add coverage for ansible config report

### PR DESCRIPTION
**Pull request by adarshdubey-star**
Test results:

```
pytest tests/foreman/ui/test_ansible.py -k test_positive_config_report_ansible
======================================================================================= test session starts =======================================================================================
platform linux -- Python 3.8.6, pytest-7.1.3, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: forked-1.4.0, services-2.2.1, cov-3.0.0, xdist-3.1.0, ibutsu-2.2.4, reportportal-5.1.3, mock-3.10.0
collected 5 items / 3 deselected / 2 selected                                                                                                                                                     

tests/foreman/ui/test_ansible.py ..                                                                                                                                                         [100%]
==================================================================== 2 passed, 3 deselected, 3 warnings in 1363.22s (0:22:43) =====================================================================

```
Airgun PR for the endpoints utilized : https://github.com/SatelliteQE/airgun/pull/791

**Note**: `The results are from 6.12 we have a issue in 6.13 hence going for the skip marker`
BZ : https://bugzilla.redhat.com/show_bug.cgi?id=2155814

### CODE REVIEW

File: tests/foreman/ui/test_ansible.py [hunk 1]
```diff
@@ -115,3 +118,65 @@ def test_positive_create_variable_with_overrides(target_sat):
             }
         )
         assert session.ansiblevariables.search(key)[0]['Name'] == key
+
+
+@pytest.mark.skip_if_open("BZ:2155814")
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:123]:
```suggestion
@pytest.mark.skip_if_open('BZ:2155814')
```

File: tests/foreman/ui/test_ansible.py [hunk 2]
```diff
@@ -115,3 +118,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_list([8, 9])
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:125]: Is it possible to include EL7 and fips hosts?

File: tests/foreman/ui/test_ansible.py [hunk 3]
```diff
@@ -115,3 +118,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+def test_positive_config_report_ansible(session, target_sat, module_org, rhel_contenthost):
+    """
+    Test Config Report generation with Ansible Jobs.
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:128]:
```suggestion
    """Test Config Report generation with Ansible Jobs.
```

File: tests/foreman/ui/test_ansible.py [hunk 4]
```diff
@@ -115,3 +118,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+        3. Assign that role to a host
+        4. Assert that the role was assigned to the host successfully
+        5. Run the Ansible playbook associated with that role
+        6. Check if the report is created successfully
+
+    :expectedresults:
+        1. Host should be assigned the proper role.
+        2. Job report should be created.
+
+    :CaseAutomation: Automated
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:144]: CaseAutomation already defined at module level

File: tests/foreman/ui/test_ansible.py [hunk 5]
```diff
@@ -115,3 +118,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    rhel_contenthost.install_katello_ca(target_sat)
+    rhel_contenthost.register_contenthost(module_org.label, force=True)
+    assert rhel_contenthost.subscribed
+    rhel_contenthost.add_rex_key(satellite=target_sat)
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:150]: can we use global registration here instead? non-blocking

File: tests/foreman/ui/test_ansible.py [hunk 6]
```diff
@@ -115,3 +118,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+            entity_name='Run ansible roles', host_name=rhel_contenthost.hostname
+        )
+        status = session.jobinvocation.read(
+            entity_name='Run ansible roles', host_name=rhel_contenthost.hostname
+        )
+        assert status['overview']['hosts_table'][0]['Status'] == 'success'
+        session.configreport.search(rhel_contenthost.hostname)
+        session.configreport.delete(rhel_contenthost.hostname)
+        status = session.configreport.read()
+        assert len(status['table']) == 0
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:182]:
```suggestion
        assert len(session.configreport.read()['table']) == 0
```

File: tests/foreman/ui/test_ansible.py [hunk 7]
```diff
@@ -115,3 +118,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+        )
+        session.jobinvocation.wait_job_invocation_state(
+            entity_name='Run ansible roles', host_name=rhel_contenthost.hostname
+        )
+        status = session.jobinvocation.read(
+            entity_name='Run ansible roles', host_name=rhel_contenthost.hostname
+        )
+        assert status['overview']['hosts_table'][0]['Status'] == 'success'
+        session.configreport.search(rhel_contenthost.hostname)
+        session.configreport.delete(rhel_contenthost.hostname)
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:180]: Not sure if the config report gets generated if job-invocation fails, so can this delete call be moved to teardown with a request.addfinalizer?

File: tests/foreman/ui/test_ansible.py (hunk 2 above)
Comment (adarshdubey-star) [reply tests/foreman/ui/test_ansible.py:125]: The Role which I am using is related to insights-client and that package is not present on rhel 7 so didn't include that. I need to explicitly install that on the VM then I think include that.

File: tests/foreman/ui/test_ansible.py (hunk 4 above)
Comment (adarshdubey-star) [reply tests/foreman/ui/test_ansible.py:144]: At the test level, it helps in bifurcating the manual vs automated scenarios

File: tests/foreman/ui/test_ansible.py (hunk 5 above)
Comment (adarshdubey-star) [reply tests/foreman/ui/test_ansible.py:150]: Yeah, we can but I don't see much difference in using that.

File: tests/foreman/ui/test_ansible.py (hunk 7 above)
Comment (adarshdubey-star) [reply tests/foreman/ui/test_ansible.py:180]: Good catch! Yes if the Job gets failed then also the report will be generated.

File: tests/foreman/ui/test_ansible.py (hunk 2 above)
Comment (Gauravtalreja1) [reply tests/foreman/ui/test_ansible.py:125]: @adarshdubey-star I guess that role is just an example here, so i wouldn't skip el7 just for that purpose, can you try using any other role instead where we can test all versions?

File: tests/foreman/ui/test_ansible.py (hunk 4 above)
Comment (Gauravtalreja1) [reply tests/foreman/ui/test_ansible.py:144]: yes, but if something is defined at module level so we don't repeat them at test level, so this can be removed

File: tests/foreman/ui/test_ansible.py (hunk 5 above)
Comment (Gauravtalreja1) [reply tests/foreman/ui/test_ansible.py:150]: With global registration, you won't need explicit steps for install_katello_ca and add_rex_key

File: tests/foreman/ui/test_ansible.py [hunk 8]
```diff
@@ -115,3 +119,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+        rhel_contenthost.create_custom_repos(rhel7=settings.repos.rhel7_os)
+        assert rhel_contenthost.execute('yum install -y insights-client').status == 0
+    rhel_contenthost.install_katello_ca(target_sat)
+    rhel_contenthost.register_contenthost(module_org.label, force=True)
+    assert rhel_contenthost.subscribed
+    rhel_contenthost.add_rex_key(satellite=target_sat)
+    id = target_sat.internal_capsule.id
+    target_host = target_sat.api.Host().search(
+        query={'search': f'name={rhel_contenthost.hostname}'}
+    )
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:156]: can you try `target_host = rhel_contenthost.nailgun_host` and `target_host.assign_ansible_roles()` here instead?

File: tests/foreman/ui/test_ansible.py [hunk 9]
```diff
@@ -115,3 +119,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+        6. Check if the report is created successfully
+
+    :expectedresults:
+        1. Host should be assigned the proper role.
+        2. Job report should be created.
+
+    :BZ: 2155814
+    """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    if rhel_contenthost.os_version.major == 7:
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:146]: I know we're skipping el6, but just to be explicit  
```suggestion
    if rhel_contenthost.os_version.major <= 7:
```

File: tests/foreman/ui/test_ansible.py [hunk 10]
```diff
@@ -115,3 +119,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+        rhel_contenthost.create_custom_repos(rhel7=settings.repos.rhel7_os)
+        assert rhel_contenthost.execute('yum install -y insights-client').status == 0
+    rhel_contenthost.install_katello_ca(target_sat)
+    rhel_contenthost.register_contenthost(module_org.label, force=True)
+    assert rhel_contenthost.subscribed
+    rhel_contenthost.add_rex_key(satellite=target_sat)
+    id = target_sat.internal_capsule.id
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:153]: Replacing `internal_capsule`with `nailgun_smart_proxy` property in Capsule hosts class, if this PR https://github.com/SatelliteQE/robottelo/pull/10458 gets in first, then we'll need to update this.

Comment (adarshdubey-star) [reply tests/foreman/ui/test_ansible.py:153]: Yes

File: tests/foreman/ui/test_ansible.py (hunk 8 above)
Comment (adarshdubey-star) [reply tests/foreman/ui/test_ansible.py:156]: Sure will try this.

File: tests/foreman/ui/test_ansible.py [hunk 11]
```diff
@@ -115,3 +119,65 @@ def test_positive_create_variable_with_overrides(target_sat):
...
+        3. Assign that role to a host
+        4. Assert that the role was assigned to the host successfully
+        5. Run the Ansible playbook associated with that role
+        6. Check if the report is created successfully
+
+    :expectedresults:
+        1. Host should be assigned the proper role.
+        2. Job report should be created.
+
+    :BZ: 2155814
```
Comment (Gauravtalreja1) [inline tests/foreman/ui/test_ansible.py:143]: can we remove this BZ as it seems unrelated?

File: tests/foreman/ui/test_ansible.py (hunk 8 above)
Comment (adarshdubey-star) [reply tests/foreman/ui/test_ansible.py]: Works well @Gauravtalreja1 , thanks for the suggestion.

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py -k test_positive_config_report_ansible

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py -k test_positive_config_report_ansible
nailgun: 841

**Comment by adarshdubey-star** [comment]
Last two failures happening because of : https://bugzilla.redhat.com/show_bug.cgi?id=2158565

File: tests/foreman/ui/test_ansible.py (hunk 10 above)
Comment (Gauravtalreja1) [reply tests/foreman/ui/test_ansible.py:152]: @adarshdubey-star Above PR is merged, you can update this

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py -k test_positive_config_report_ansible

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py -k test_positive_config_report_ansible
nailgun: 881

**Comment by adarshdubey-star** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_ansible.py -k test_positive_config_report_ansible
nailgun: 881

**Review by shubhamsg199 — APPROVED** [review]
ACK

**Review by Gauravtalreja1 — APPROVED** [review]
